scene/dictionary-renderer-graphics-overlay/src/main/java/com/esri/samples/scene/DictionaryRendererGraphicsOverlay.java

```diff
@@ -0,0 +1,221 @@
+    Map<String, Object> attributes = new HashMap<>();
+    try {
+      File symbolData = new File(getClass().getResource("/Mil2525DMessages.xml").getPath());
+      FileInputStream file = new FileInputStream(symbolData.getAbsolutePath());
+      XMLStreamReader xmlReader = XMLInputFactory.newInstance().createXMLStreamReader(file);
+
+      // skip first element tag
+      xmlReader.next();
+      while (xmlReader.hasNext()) {
+        // if next thing read is an element from the xml file
+        if (xmlReader.next() == XMLStreamConstants.START_ELEMENT) {
+          String attributeName = xmlReader.getLocalName().trim();
+          if (attributeName.equals("message") && attributes.size() > 0) {
+            createGraphic(attributes);
+            attributes = new HashMap<>();
+          } else {
+            xmlReader.next();
+            String attributeValue = xmlReader.getText().trim();
+            attributes.put(attributeName, attributeValue);
+          }
+        }
+      }
+      // create graphic for last block of attributes
+      createGraphic(attributes);
+      if (file != null)
+        file.close();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Creates a graphic using a symbol dictionary and the attributes that were passed.
+   * 
+   * @param attributes tells symbol dictionary what symbol to apply to graphic
+   */
+  private void createGraphic(Map<String, Object> attributes) {
+
+    Geometry geometry = null;
+    int wkid = Integer.parseInt((String) attributes.get("_wkid"));
+    SpatialReference sr = SpatialReference.create(wkid);
+    // get points that make up the graphic's geometry
+    String[] geometryPoints = ((String) attributes.get("_control_points")).split(";");
+
+    // geometry is made up of a single point
+    if (geometryPoints.length == 1) {
+      // split into x,y coordinates
+      String[] coordinates = geometryPoints[0].split(",");
+      geometry = new Point(Double.parseDouble(coordinates[0]), Double.parseDouble(coordinates[1]), sr);
+    } else {
+      geometry = createBuilderFromPoints(geometryPoints, sr).toGeometry();
+    }
+
+    if (!geometry.isEmpty()) {
+      Graphic graphic = new Graphic(geometry, attributes);
+      graphicsOverlay.getGraphics().add(graphic);
+
+      // update boundary geometry so all graphics are within
+      graphicsBoundary = graphicsBoundary == null ? geometry : GeometryEngine.union(graphicsBoundary, geometry);
+    }
+  }
+
+  /**
+   * Builds a polygon or polyline from the points that are passed.
+   * 
+   * @param points used to build a polygon or polyline geometry
+   * @param sr spatial reference of the geometry to be built
+   * @return geometry that was built
+   */
+  private MultipartBuilder createBuilderFromPoints(String[] points, SpatialReference sr) {
+
+    MultipartBuilder builder = null;
+    if (points.length >= 3 && points[0] == points[points.length - 1]) {

```

Comment:
this test never passes since == is for object (String instance) equality, you could use
points[0].contentEquals(points[points.length - 1])